src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -485,4 +492,46 @@ else if (participation instanceof ProgrammingExerciseParticipation) {
         final var result = resultService.createNewExampleResultForSubmissionWithExampleSubmission(submissionId, isProgrammingExerciseWithFeedback);
         return new ResponseEntity<>(result, HttpStatus.CREATED);
     }
+
+    @PostMapping(value = "/exercises/{exerciseId}/external-submission-results")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Result> createResultForExternalSubmission(@PathVariable Long exerciseId, @RequestParam String studentLogin, @RequestBody Result result)
+            throws URISyntaxException {
+        log.debug("REST request to create Result for External Submission for Exercise : {}", exerciseId);
+
+        Exercise exercise = exerciseService.findOneWithAdditionalElements(exerciseId);
+        User user = userService.getUserWithGroupsAndAuthorities();
+        Optional<User> student = userService.getUserWithAuthoritiesByLogin(studentLogin);
+        Course course = exercise.getCourse();
+
+        if (!authCheckService.isAtLeastInstructorForExercise(exercise, user)) {
+            throw new AccessForbiddenException("You are not allowed to access this resource");
+        }
+        if (student.isEmpty() || !authCheckService.isAtLeastStudentInCourse(course, student.get())) {
+            throw new ResponseStatusException(HttpStatus.FAILED_DEPENDENCY, "No student found for " + studentLogin + " in exercise " + exerciseId);
+        }
+        if (exercise instanceof QuizExercise) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "External submissions are not supported for Quiz exercises.");
+        }
+
+        StudentParticipation participation = participationService.createParticipationWithEmptySubmission(exercise, student.get(), SubmissionType.EXTERNAL);
+        result.setParticipation(participation);
+
+        participation = (StudentParticipation) participationService.findOneWithEagerSubmissions(participation.getId());
+        Optional<Submission> optionalSubmission = participation.findLatestSubmission();
+        if (optionalSubmission.isEmpty()) {
```

Comment:
Can this case really happen that the latest submission is empty?

Reply:
No, it should not happen. If that were the case, an exception should have already been thrown previously. I wasn't sure if just assuming the Optional to exist is bad style but I guess checking if it is empty could also cause confusion for other developers since this is not an expected state.